TPA-85: Vault password encryption with keyring

JonathanRenon-EDB opened the pull request:
Change to TPA vault password default storage to use system keyring.
currently TPA stores vault password as plain text file, this work leverage python keyring to store this vault password more securely via system keyring such as secret-service.

this also include changes to configure help output.

JonathanRenon-EDB committed:
Modify architecture parser layout

Reorder options into more meaningful groups.

Improve readability of `--help` and consistently use help text starting
with a lowercase letter.

JonathanRenon-EDB committed:
Store the ansible-vault password in the system keyring by default

We used to write the vault password to vault/vault_pass.txt, and we'll
continue to work with existing clusters where that is the case. For new
clusters, we'll try to use the keyring module to store the password into
any available system keyring (e.g., gnome-keyring).

Use `tpaexec configure … --keyring-backend <name>` to select a backend,
but the only available choices are "system" (the default) and "legacy"
(which is to use vault/vault_pass.txt).

Modify use-vault to avoid having to copy it over in cluster_dirs
leverage the fact that tpaexec script `cd` to cluster_dir
before running ansible commands.

References: TPA-85

JonathanRenon-EDB commented:
rebased today, starting to look into the discussion we had with Haroon about streamlining the experience for existing clusters.
atm the new required script use-vault is pushed via provision but the step can be skipped if no changes are detected in config.yml (feature that was added after initial dev for tpa-85).
we need to find a better way to add this script without user interaction required.

JonathanRenon-EDB committed:
Add RELNOTES.md entry for TPA-85

haroon-github reviewed (changes requested):
With use-vault, we will need to consider the scenario where commands, for example `download-packages` and `deprovision`, are run on clusters that have not yet been provisioned and therefore no inventory exists. At the moment, current implementation will error out for such scenarios. Here for example,

```
~/cluster/xxx_yy$ tpaexec download-packages . --docker-image tpa/ubuntu:22.04 -vvv
ansible-playbook 2.9.27
  config file = /home/haroon/tpa/ansible/ansible.cfg
  configured module search path = ['/home/haroon/tpa/library']
  ansible python module location = /home/haroon/tpa/tpa-venv/lib/python3.10/site-packages/ansible
  executable location = /home/haroon/tpa/tpa-venv/bin/ansible-playbook
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Using /home/haroon/tpa/ansible/ansible.cfg as config file
Could not find vault password in system keyring, please use --ask-vault-pass
or ensure entry:
service: TPA_xxx_yy
username: vault_pass
exists with the correct vault password using:
`keyring set TPA_xxx_yy vault_pass`
[WARNING]: Error in vault password file loading (default): Vault password script /home/haroon/tpa/architectures/lib/use-vault returned non-zero (1): None
ERROR! Vault password script /home/haroon/tpa/architectures/lib/use-vault returned non-zero (1): None
```

JonathanRenon-EDB committed:
Fix error when no vault_pass exist

Avoid using --vault-password-file option when vault_pass has not been
generated yet (no provision)

the only time we don't test first is for store-password and show-password. these commands don't make sense before having a vault_pass.

slight change in the error message displayed by show-password function.

JonathanRenon-EDB committed:
fixup

JonathanRenon-EDB commented:
closing this PR
